core/io/resource_format_binary.cpp

```diff
@@ -1814,11 +1814,11 @@ Error ResourceFormatSaverBinaryInstance::save(const String &p_path, const RES &p
 
 				if (skip_editor && F->get().name.begins_with("__editor"))
 					continue;
-				if (F->get().usage & PROPERTY_USAGE_STORAGE) {
+				if ((F->get().usage & PROPERTY_USAGE_STORAGE)) {
```

Comment:
Why are the parentheses needed here?

Reply:
A 'suggestion' warning for GCC about comparing things in a binary context. I can remove the ones that are completely unambiguous to humans and disable the warning later though.

Reply:
I would prefer to keep the ones where its 'if (stuff & blah && something else)' as the single '&' and the double '&&' actually could be mistaken for each other.

Reply:
Well of course it makes sense if `(stuff & blah) && blabla`, but here it's unambiguous so it seems weird that GCC would raise a warning. But if it does, I guess the extraneous parentheses don't hurt. I'm just pretty sure that they're useless as whoever adds a new conditional there might just do `if ((F->get().usage & PROPERTY_USAGE_STORAGE && new_cond))` :P

Reply:
Yeah, this one will require some upkeep probably. I don't mind doing this manually from time to time. I think we'll never be able to enable -Werror indiscriminately. I'm hoping we can get a subset of warnings always enabled with -Werror by default though. maybe unitialized, unused, set but unused, misleading indendtation. 

We'll discuss that when I make a PR to take a first stab at that.